src/sentry/static/sentry/app/views/settings/project/filtersAndSampling/rules/index.tsx

```diff
@@ -3,36 +3,118 @@ import styled from '@emotion/styled';
 
 import {PanelTable} from 'app/components/panels';
 import {t} from 'app/locale';
-import overflowEllipsis from 'app/styles/overflowEllipsis';
 import {DynamicSamplingRule} from 'app/types/dynamicSampling';
 
+import DragHandle from './dragHandle';
 import Rule from './rule';
+import {layout} from './utils';
 
 type Props = {
   rules: Array<DynamicSamplingRule>;
+  disabled: boolean;
   onEditRule: (rule: DynamicSamplingRule) => () => void;
   onDeleteRule: (rule: DynamicSamplingRule) => () => void;
-  disabled: boolean;
+  onUpdateRules: (rules: Array<DynamicSamplingRule>) => void;
+};
+
+type RulesWithId = Array<DynamicSamplingRule & {id: string}>;
+
+type State = {
+  rules: RulesWithId;
 };
 
-function Rules({rules, onEditRule, onDeleteRule, disabled}: Props) {
-  return (
-    <StyledPanelTable
-      headers={['', t('Event Type'), t('Category'), t('Sampling Rate'), '']}
-      isEmpty={!rules.length}
-      emptyMessage={t('There are no rules to display')}
-    >
-      {rules.map((rule, index) => (
-        <Rule
-          key={index}
-          rule={rule}
-          onEditRule={onEditRule(rule)}
-          onDeleteRule={onDeleteRule(rule)}
-          disabled={disabled}
+class Rules extends React.PureComponent<Props, State> {
+  state: State = {
+    rules: [],
+  };
+
+  componentDidMount() {
+    this.getRules();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.rules !== this.props.rules) {
```

Comment:
Do you need any deeper comparison here as `rules` is an array?

Reply:
Yes ... I was thinking of using the  `isEqual` function of `lodash` but until now everything worked out of the box...